source/models/table_metadata.py

```diff
@@ -18,13 +18,18 @@ def get_path(self):
         return self.table.get_path()
 
     def serialize_table_shcema(self):
-        table_schemafile_path =  os.path.join(self.get_path(), "schema.json")
+        table_schemafile_path =  os.path.join(self.get_path(), "schema.json") 
         with open(table_schemafile_path, 'w') as table_schema_file:
             json.dump(self.table_schema, table_schema_file,indent=2)
+        self.serialize_folders()
         self.serialize_indecies()
 
     def serialize_indecies(self):
-        os.makedirs(os.path.join(self.get_path(),"indices"), exist_ok = True)
-        for index_value in self.index_keys: 
-            index = Index(table = self.table, index_value = index_value)
-            index.serialize()
\ No newline at end of file
+        os.makedirs(os.path.join(self.get_path(),"Indices"), exist_ok = True)
+        for index_name in self.index_keys: 
+            index = Index(table = self.table, index_name = index_name)
+            index.serialize()
+
+    def serialize_folders(self):
```

Comment:
You are creating folder. Serialize mean you convert an object in memory to text in file

Reply:
Make it private